lib/rucio/tests/test_cache.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from rucio.common.cache import (ENABLE_CACHING, ignore_arguments_key_generator,
+                                make_region_memcached)
+from rucio.db.sqla.session import (get_session, read_session, stream_session,
+                                   transactional_session)
+
+no_ignored_arguments_function_key_generator = ignore_arguments_key_generator([])
+
+
+class TestBasicCacheKeyGenerator:
+    @pytest.mark.parametrize(
+        "left,str_comp_func,right",
+        [
+            ([], str.__eq__, []),
+            ([None], str.__eq__, [None]),
```

Comment:
Check out the `operator` module.

Reply:
Change in functionality, this part got removed